packages/gatsby-plugin-utils/src/polyfill-remote-file/utils/url-generator.ts

```diff
@@ -1,28 +1,60 @@
+import { basename, extname } from "path"
+import { createContentDigest } from "gatsby-core-utils/create-content-digest"
 import { isImage } from "../types"
 import type { ImageCropFocus, WidthOrHeight } from "../types"
 
-export function generatePublicUrl(
-  {
-    url,
-    mimeType,
-  }: {
-    url: string
-    mimeType: string
-  },
-  checkMimeType: boolean = true
+// export enum ImageCDNKeys {
+//   url: 'u',
+
+// }
+
+export function generateFileUrl({
+  url,
+  filename,
+}: {
+  url: string
+  filename: string
+}): string {
+  const fileExt = extname(filename)
+  const filenameWithoutExt = basename(filename, fileExt)
+
+  return `${generatePublicUrl({ url })}/${encodeURIComponent(
+    filenameWithoutExt
+  )}${fileExt}`
+}
+
+export function generateImageUrl(
+  source: { url: string; mimeType: string; filename: string },
+  imageArgs: Parameters<typeof generateImageArgs>[0]
 ): string {
-  const remoteUrl = Buffer.from(url).toString(`base64`)
+  const filenameWithoutExt = basename(source.filename, extname(source.filename))
+
+  return `${generatePublicUrl(source)}/${createContentDigest(
+    generateImageArgs(imageArgs)
+  )}/${encodeURIComponent(filenameWithoutExt)}.${
+    imageArgs.format
+  }?u=${encodeURIComponent(source.url)}&a=${encodeURIComponent(
```

Comment:
We'll need to update our router code to support this new structure and the old one before releasing this